iox_query/src/physical_optimizer/sort/parquet_sortness.rs

```diff
@@ -0,0 +1,448 @@
+
+    #[test]
+    fn test_parquet_has_different_ordering() {
+        let schema = schema();
+        let base_config = FileScanConfig {
+            object_store_url: ObjectStoreUrl::parse("test://").unwrap(),
+            file_schema: Arc::clone(&schema),
+            file_groups: vec![vec![file(1), file(2)]],
+            statistics: Statistics::default(),
+            projection: None,
+            limit: None,
+            table_partition_cols: vec![],
+            output_ordering: Some(ordering(["col1", "col2"], &schema)),
+            infinite_source: false,
+        };
+        let inner = ParquetExec::new(base_config, None, None);
+        let plan = Arc::new(
+            SortExec::try_new(
+                ordering(["col2", "col1"], &schema),
+                Arc::new(inner),
+                Some(42),
+            )
+            .unwrap(),
+        );
+        let opt = ParquetSortness::default();
+        insta::assert_yaml_snapshot!(
+            OptimizationTest::new(plan, opt),
+            @r###"
+        ---
+        input:
+          - " SortExec: fetch=42, expr=[col2@1 ASC,col1@0 ASC]"
+          - "   ParquetExec: limit=None, partitions={1 group: [[1.parquet, 2.parquet]]}, projection=[col1, col2, col3]"
+        output:
+          Ok:
+            - " SortExec: fetch=42, expr=[col2@1 ASC,col1@0 ASC]"
+            - "   ParquetExec: limit=None, partitions={1 group: [[1.parquet, 2.parquet]]}, projection=[col1, col2, col3]"
+        "###
+        );
+    }
+
+    #[test]
+    fn test_parquet_has_no_ordering() {
+        let schema = schema();
+        let base_config = FileScanConfig {
+            object_store_url: ObjectStoreUrl::parse("test://").unwrap(),
+            file_schema: Arc::clone(&schema),
+            file_groups: vec![vec![file(1), file(2)]],
+            statistics: Statistics::default(),
+            projection: None,
+            limit: None,
+            table_partition_cols: vec![],
+            output_ordering: None,
+            infinite_source: false,
+        };
+        let inner = ParquetExec::new(base_config, None, None);
+        let plan = Arc::new(
+            SortExec::try_new(
+                ordering(["col2", "col1"], &schema),
+                Arc::new(inner),
+                Some(42),
+            )
+            .unwrap(),
+        );
+        let opt = ParquetSortness::default();
+        insta::assert_yaml_snapshot!(
+            OptimizationTest::new(plan, opt),
+            @r###"
+        ---
+        input:
+          - " SortExec: fetch=42, expr=[col2@1 ASC,col1@0 ASC]"
+          - "   ParquetExec: limit=None, partitions={1 group: [[1.parquet, 2.parquet]]}, projection=[col1, col2, col3]"
+        output:
+          Ok:
+            - " SortExec: fetch=42, expr=[col2@1 ASC,col1@0 ASC]"
+            - "   ParquetExec: limit=None, partitions={1 group: [[1.parquet, 2.parquet]]}, projection=[col1, col2, col3]"
+        "###
+        );
+    }
+
+    #[test]
+    fn test_limit() {
```

Comment:
```suggestion
    fn test_fanout_limit() {
```

I got confused initially that this test had to do with the `limit: None` on the FileScanConfig